libraries/SITL/SIM_GPS.cpp

```diff
@@ -1530,6 +1531,7 @@ void GPS::update()
 
         // run at configured GPS rate (default 5Hz)
         if ((now_ms - last_update) < (uint32_t)(1000/_sitl->gps_hertz[idx])) {
+            backend->update_read();
```

Comment:
Why is this inside the `if`?

For that matter, you're leaving the `update_write` to be still dependent on the GPS update rate - but a real device is going to respond to your configuration requests outside a 5Hz result update rate.

Ideally we get a configuration directive at time <t>, stash that away and respond after, say, 20ms with the response, that sort of thing.  But the scheduled messages should go out at their normal rates (assuming this is what the real devices do, which I think is a reasonable assumption).

Reply:
Yes, the goal of this PR is to not affect physics or write speed; merely to allow reasonably quick reading. THat's what I implemented in my PR - configuration responses are nearly instant, and decoupled from periodic streaming.

Trimble GSOF responds within 3mS; I did this to service the reads as fast as possible. Why queue and respond after 20mS? GSOF responds to all configuration as soon as it can parse, regardless if you send one or multiple at a time.

Reply:
> Trimble GSOF responds within 3mS; I did this to service the reads as fast as possible. Why queue and respond after 20mS? GSOF responds to all configuration as soon as it can parse, regardless if you send one or multiple at a time.

20ms was just an example.  3ms would be the appropriate delay.

"update_read" should *not* be writing data back to the autopilot.  That's what "update_write" should be doing.  IIRC that's problematic in there for some reason or another.

Reply:
> > Trimble GSOF responds within 3mS; I did this to service the reads as fast as possible. Why queue and respond after 20mS? GSOF responds to all configuration as soon as it can parse, regardless if you send one or multiple at a time.
> 
> 20ms was just an example. 3ms would be the appropriate delay.
> 
> "update_read" should _not_ be writing data back to the autopilot. That's what "update_write" should be doing. IIRC that's problematic in there for some reason or another.

It's working on my branch, it's all on the same thread,sequential and only called from one spot in the front end. 

The way update_write was written is to always form packets and write them out every time it's called. Do you have a recommendation? I was trying to avoid adding extra buffers or invasive changes.

Reply:
I think it is the names that are the issue. If update_write() was called update_frame() or update_publish() and it did the 5 Hz (or configured rate) sending of the frames, then a update_fast() method is for backends that need to do things in-between, and update_fast() would be empty by default

Reply:
Change `update_write` to `publish` and remove the extra comment, then ping peter.